### ISSUE TITLE
Slow Searching and downloading

### ISSUE THREAD

**Issue by samuelpetrosyan** [deepjyoti30/ytmdl#190]
# Bug report

## Version Info

- Ytmdl version: `latest`
- Youtube-dl version: `latest`

## How Ytmdl was installed?

Write here how you installed ytmdl. If it was through pip or some other way.

pip install ytmdl

## Description

It takes about 1 minute to get the result of music searching. And then downloading also takes a long time! Is there any solution to this problem??

Thanks in advance and also for a good tool!!

**Comment by deepjyoti30** [comment]
@samuelpetrosyan The latest version of ytmdl uses youtube-dl under the hood so if downloading is slow it might be something with your connection.

Can you try downloading the same song with youtube-dl and see if it's the same.

Regarding searching, I'll take a look at that, might be that the library being used `youtube-search` is not working as expected.

**Comment by samuelpetrosyan** [comment]
Yeap, I also researched and noticed that the problem is in youtube-search. And interesting that on one node it does search so fast and on another node it does it so slow! So much contrast...
Maybe it's an youtube search design issue. Anyways thanks.

**Comment by deepjyoti30** [comment]
@samuelpetrosyan Yeah, I will take a look to see if the issue is happening on a lot of songs. Can you perhaps share some name of songs that `youtube-search` took time to find?

**Comment by samuelpetrosyan** [comment]
> @samuelpetrosyan Yeah, I will take a look to see if the issue is happening on a lot of songs. Can you perhaps share some name of songs that `youtube-search` took time to find?

song name did not matter.

**Comment by deepjyoti30** [comment]
@samuelpetrosyan I faced the issue too.

Turns out the first search is pretty fast with `youtube-search` but the second search takes forever. That's not convenient. I made some changes to test with [youtube-search-python](https://github.com/alexmercerind/youtube-search-python) instead and it's working like a charm.

I have noticed that with the latter the search takes about a second more than the one before but at least it's consistent within searches!
